protocol/errors.go

```diff
@@ -46,6 +46,8 @@ const (
 	CodeInvalidTxnError ErrorCode = 22
 	//CodeAddTxnError is returned when adding txn to state db fails
 	CodeAddTxnError ErrorCode = 23
+	//CodeAddKeyPageIndices is returned when fetching key indices fail
+	CodeAddKeyPageIndices ErrorCode = 24
```

Comment:
Why is this "Add"? Regardless, I think I would prefer returning `CodeNotFound` if the key is not found.